python/cugraph/cugraph/dask/link_analysis/pagerank.py

```diff
@@ -13,65 +13,131 @@
 # limitations under the License.
 #
 
-from dask.distributed import wait, default_client
-from cugraph.dask.common.input_utils import (get_distributed_data,
-                                             get_vertex_partition_offsets)
-from cugraph.dask.link_analysis import mg_pagerank_wrapper as mg_pagerank
+from dask.distributed import wait
 import cugraph.dask.comms.comms as Comms
 import dask_cudf
-from dask.dataframe.shuffle import rearrange_by_column
-
-
-def call_pagerank(sID,
-                  data,
-                  src_col_name,
-                  dst_col_name,
-                  num_verts,
-                  num_edges,
-                  vertex_partition_offsets,
-                  aggregate_segment_offsets,
-                  alpha,
-                  max_iter,
-                  tol,
-                  personalization,
-                  nstart):
-    wid = Comms.get_worker_id(sID)
-    handle = Comms.get_handle(sID)
-    local_size = len(aggregate_segment_offsets) // Comms.get_n_workers(sID)
-    segment_offsets = \
-        aggregate_segment_offsets[local_size * wid: local_size * (wid + 1)]
-    return mg_pagerank.mg_pagerank(data[0],
-                                   src_col_name,
-                                   dst_col_name,
-                                   num_verts,
-                                   num_edges,
-                                   vertex_partition_offsets,
-                                   wid,
-                                   handle,
-                                   segment_offsets,
-                                   alpha,
-                                   max_iter,
-                                   tol,
-                                   personalization,
-                                   nstart)
+import cudf
+import numpy as np
+import warnings
+from cugraph.dask.common.input_utils import get_distributed_data
 
+from pylibcugraph import (ResourceHandle,
+                          pagerank as pylibcugraph_pagerank,
+                          personalized_pagerank as pylibcugraph_p_pagerank
+                          )
 
-def pagerank(input_graph,
-             alpha=0.85,
-             personalization=None,
-             max_iter=100,
-             tol=1.0e-5,
-             nstart=None):
 
+def convert_to_cudf(cp_arrays):
+    """
+    Creates a cudf DataFrame from cupy arrays from pylibcugraph wrapper
+    """
+    cupy_vertices, cupy_pagerank = cp_arrays
+    df = cudf.DataFrame()
+    df["vertex"] = cupy_vertices
+    df["pagerank"] = cupy_pagerank
+
+    return df
+
+
+def ensure_valid_dtype(input_graph, input_df, input_df_name):
```

Comment:
Can you add a FIXME saying this should be moved to a utility module to be shared by both other algos (especially SG and MG)?